modules/coursier/shared/src/main/scala/coursier/Resolve.scala

```diff
@@ -215,12 +215,18 @@ object Resolve extends PlatformResolve {
         val scalaOrg =
           if (params.typelevel) Organization("org.typelevel")
           else Organization("org.scala-lang")
-        Seq(
-          Module(scalaOrg, ModuleName("scala-library")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scala-reflect")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scala-compiler")) -> params.selectedScalaVersion,
-          Module(scalaOrg, ModuleName("org.scala-lang:scalap")) -> params.selectedScalaVersion
-        )
+        if (params.selectedScalaVersion.startsWith("3"))
+          Seq(
+            Module(scalaOrg, ModuleName("scala3-library")) -> params.selectedScalaVersion,
+            Module(scalaOrg, ModuleName("org.scala-lang:scala3-compiler")) -> params.selectedScalaVersion
```

Comment:
I think the former code was wrong 😅, the organization shouldn't be here:
```suggestion
            Module(scalaOrg, ModuleName("scala3-compiler")) -> params.selectedScalaVersion
```

Reply:
Glad you've noticed that; I was a bit wondered but thought that it's a part of some plan :smile: